### PR TITLE
Locale save by change local switch

### LINKED ISSUE #1929: 🚀 [Feature] When registering the user the set language must be saved
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-16T13:30:06Z" title="Wednesday, October 16th 2019, 3:30:06 pm +02:00">Oct 16, 2019</time>_
_Closed <time datetime="2019-10-24T08:02:47Z" title="Thursday, October 24th 2019, 10:02:47 am +02:00">Oct 24, 2019</time>_
---

## :rocket: Feature
when registering the user the set language must be saved

  
- create a new value for the user in the database. 
  
-  save the  setting of the language in your new value when registering. 

### group
dieses Problem ist ein Teil von https://github.com/Human-Connection/Human-Connection/issues/1931

### PR DESCRIPTION
**Pull request by Tirokk**
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-22T09:37:39Z" title="Tuesday, October 22nd 2019, 11:37:39 am +02:00">Oct 22, 2019</time>_
_Merged <time datetime="2019-10-24T08:02:46Z" title="Thursday, October 24th 2019, 10:02:46 am +02:00">Oct 24, 2019</time>_
---

## 🍰 Pullrequest
i have attached an apollo.mutation to the language selection. which saves the language for the user in the database. 

### Issues
- fixes #1929

### CODE REVIEW

File: webapp/components/LocaleSwitch/LocaleSwitch.vue [hunk 1]
```diff
@@ -64,15 +66,44 @@ export default {
...
       return routes
     },
+    ...mapGetters({
+      currentUser: 'auth/user',
+    }),
   },
   methods: {
     changeLanguage(locale, toggleMenu) {
       this.$i18n.set(locale)
+      this.updateUserLocale()
```
Comment (Tirokk) [inline webapp/components/LocaleSwitch/LocaleSwitch.vue:76]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-24T06:44:03Z" title="Thursday, October 24th 2019, 8:44:03 am +02:00">Oct 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -64,15 +66,44 @@ export default {
       })
       return routes
     },
+    ...mapGetters({
+      currentUser: 'auth/user',
+    }),
   },
   methods: {
     changeLanguage(locale, toggleMenu) {
       this.$i18n.set(locale)
+      this.change()
```


I don't love this naming convention @ogerly, it doesn't tell me what the function does
shouldn't it be something like `updateUserLocale`?

File: webapp/graphql/User.js [hunk 2]
```diff
@@ -153,3 +153,14 @@ export const checkSlugAvailableQuery = gql`
...
+export const localeMutation = () => {
+  return gql`
+    mutation($id: ID!, $locale: String) {
+      UpdateUser(id: $id, locale: $locale) {
+        id
+        locale
+      }
+    }
+  `
+}
```
Comment (Tirokk) [inline webapp/graphql/User.js:166]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-24T06:44:39Z" title="Thursday, October 24th 2019, 8:44:39 am +02:00">Oct 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -153,3 +153,14 @@ export const checkSlugAvailableQuery = gql`
     }
   }
 `
+
+export const localeMutation = () => {
+  return gql`
+    mutation($id: ID!, $locale: String) {
+      UpdateUser(id: $id, locale: $locale) {
+        id
+        locale
+      }
+    }
+  `
+}
```


is there a reason we can't just use the `UpdateUser` mutation?

Comment (Tirokk) [reply webapp/graphql/User.js:166]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-24T06:49:27Z" title="Thursday, October 24th 2019, 8:49:27 am +02:00">Oct 24, 2019</time>_

---


i took the embed code as my template. i also wondered if a simple user update wouldn't work, but i just wasn't sure.

File: webapp/components/LocaleSwitch/LocaleSwitch.vue (hunk 1 above)
Comment (Tirokk) [reply webapp/components/LocaleSwitch/LocaleSwitch.vue:76]: > [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-10-24T06:50:35Z" title="Thursday, October 24th 2019, 8:50:35 am +02:00">Oct 24, 2019</time>_

---


yeah, you're right. KISS does not always make sense. ;) I change it.

File: webapp/graphql/User.js (hunk 2 above)
Comment (Tirokk) [reply webapp/graphql/User.js:166]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-24T06:50:41Z" title="Thursday, October 24th 2019, 8:50:41 am +02:00">Oct 24, 2019</time>_

---


sorry, I guess we don't have an `UpdateUser` mutation?? we have the `allowEmbedIframesMutation` in`graphql/User.js` as well, and then we have `UpdateUser` mutations scattered throughout the `webapp`....
I'd really like it if we could refactor those all into one `UpdateUser` mutation in `graphql/User.js`, but if you want to do that in a separate `PR`, I'm ok with that

**Review by Mogge — APPROVED** [review]

**Pull request closed without merging**